contracts/Options/Marginable.sol

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.17;
+
+import "../StableCoins/ERC20StableCoin.sol";
+import "../DataFeeder/ReferenceLevel.sol";
+
+abstract contract Marginable {
```

Comment:
How do we plan to use this with the optionContracts ? I see the OptionContract has a new adjustMargin call ? should the OptionContract inherit Marginable ?

Also, cannot see any tests associated with this .. you should write a 'Mock' Option contract in contracts/test/mock to exericse these methods - this is teh best way to do it as the Mock will focus on testing Marginable and not mix up the tests with the more complex terms of a contract that inherits it.